SilentUpdate/src/main/java/org/netbeans/modules/autoupdate/silentupdate/UpdateHandler.java

```diff
@@ -319,10 +350,115 @@ static Installer doVerify(InstallSupport support, Validator validator) throws Op
 
     static Restarter doInstall(InstallSupport support, Installer installer) throws OperationException
     {
+        OperationContainer<OperationSupport> con = OperationContainer.createForDirectUninstall();
         return support.doInstall(installer, null);
     }
 
-    private static boolean installNewModules()
+    static void doDisable(Collection codeNames)
+    { // codeName contains code name of modules for disable
+        Collection<UpdateElement> toDisable = new HashSet();
+        List<UpdateUnit> allUpdateUnits = UpdateManager.getDefault().getUpdateUnits(UpdateManager.TYPE.MODULE);
+        for (UpdateUnit unit : allUpdateUnits)
+        {
+            if (unit.getInstalled() != null)
```

Comment:
should change it to declaration first for clarity eg. 
``UpdateElement el = unit.getInstalled();
if (el != null)``